WordPress/gutenberg
# RadioGroup: Mark as deprecated

## What?

Mark `RadioGroup` (and `Radio`) as deprecated, so we can encourage people to use the alternatives.

And some related cleanup:

- Opt out from TypeScript checks
- Remove separate story for `Radio` (wasn't really useful anyway)
- Move `radio/` and `radio-context/` to subfolders of `radio-group/` (just to decrease clutter)

## Why?

ToggleGroupControl is visually similar and is widely used. In contrast, RadioGroup is only used in one place in the Gutenberg app, and even that is heavily modified visually. It doesn't make sense to keep this around.

<img width="280" alt="Heavily modded RadioGroup" src="https://user-images.githubusercontent.com/555336/198690403-2c5a2d14-8c90-4a8a-a4a6-66a572bf5e8f.png">

I'm not sure if we will migrate this instance and actually remove the component, but in any case it would be good to add a deprecation notice for now.

## Testing Instructions

1. `npm run storybook:dev` and see the RadioGroup story. It should've moved to the Deprecated section.
2. `npm run dev`, go to `wp-admin/site-editor.php?postType=wp_template_part`, and click the "Add New" button. The RadioGroup in the modal should not have changed.

✅ Type checks (`npx tsc --build`) still pass
